File: Sprint1/C++/D/main.cpp
// https://contest.yandex.ru/contest/22449/problems/D/
#include <iostream>

int main() {
    int n;
    std::cin >> n;

    int *temperature = new int[n];
    int chaotic = 0;

    for (int i = 0; i < n; i++) {
        std::cin >> temperature[i];
        if ((i > 1) && (temperature[i - 1] > temperature[i]) && (temperature[i - 1] > temperature[i - 2])) {
            chaotic++;
        }
    }

    if (n == 1) {
        std::cout << 1;
        return 0;
    }


    if (temperature[0] > temperature[1]) {
        chaotic++;
    }

    if (temperature[n - 1] > temperature[n - 2]) {
        chaotic++;
    }

    delete[] temperature;

    std::cout << chaotic;

    return 0;
}

File: Sprint1/D/Rust/src/main.rs
// https://contest.yandex.ru/contest/22449/problems/D/

use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let n: usize = input.trim().parse().unwrap();

    if n == 1 {
        println!("1");
        return;
    }

    input.clear();
    io::stdin().read_line(&mut input).unwrap();
    let temperatures: Vec<i32> = input
        .split_whitespace()
        .map(|s| s.parse().unwrap())
        .collect();

    let mut chaotic = 0;

    if temperatures[0] > temperatures[1] {
        chaotic += 1;
    }

    for i in 2..n {
        if temperatures[i - 1] > temperatures[i] && temperatures[i - 1] > temperatures[i - 2] {
            chaotic += 1;
        }
    }

    if temperatures[n - 1] > temperatures[n - 2] {
        chaotic += 1;
    }

    println!("{}", chaotic);
}
